components/core/tests/test-ir_encoding_methods.cpp

```diff
@@ -1284,3 +1324,59 @@ TEMPLATE_TEST_CASE(
             output_buf
     ));
 }
+
+// NOLINTNEXTLINE(readability-function-cognitive-complexity)
+TEMPLATE_TEST_CASE(
+        "ffi_ir_stream_Serializer_serialize_invalid_msgpack",
+        "[clp][ffi][ir_stream][Serializer]",
+        four_byte_encoded_variable_t,
+        eight_byte_encoded_variable_t
+) {
+    auto result{Serializer<TestType>::create()};
+    REQUIRE((false == result.has_error()));
+
+    std::stringstream msgpack_serialization_buffer;
+    auto& serializer{result.value()};
+    serializer.clear_ir_buf();
+
+    auto assert_invalid_serialization = [&](auto invalid_input) -> bool {
+        std::map<string, decltype(invalid_input)> const invalid_map{{"valid_key", invalid_input}};
```

Comment:
How about this instead?

```suggestion
    auto assert_invalid_serialization = [&]<typename T>(T invalid_input) -> bool {
        std::map<string, T> const invalid_map{{"valid_key", invalid_input}};
```